### PR TITLE
[JENKINS-57244] FilePath (de)serialization via XStream is unsupported but should not be fatal

### PR DESCRIPTION
**Pull request by jglick**
Amends #3980 to not be a fatal error. Avoids the need to back out the entire patch as in #4009. Provides one solution to a test failure I am writing in another PR.

See [JENKINS-57244](https://issues.jenkins-ci.org/browse/JENKINS-57244).

### Proposed changelog entries

* Some plugins running in Jenkins 2.175 would fail with an error related to serialization of `FilePath`s. This is now downgraded to a warning. Plugin updates should still be applied to correct the underlying mistake.

### Desired reviewers

@jenkinsci/code-reviewers

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
OK as a hotfix. Would be nice to have TODOs for changing code. Also note that XStream exceptions will be pretty useless, because users won't have means to understand from where the exception comes.

**Comment by jglick** [comment]
> TODOs for changing code

You mean, for ultimately making this fatal? I guess that could be a long way away.

> XStream exceptions will be pretty useless, because users won't have means to understand from where the exception comes

I do not know about “users” but to Jenkins developers the stack trace in each case clearly points to the culprit.

**Review by daniel-beck — DISMISSED** [review]
Plus file a corresponding PR to https://github.com/jenkins-infra/jenkins.io. Can redirect to JENKINS-57244 for a start, but an epic would probably be better, explaining the problem and linking to individual issues for plugins.

**Comment by jglick** [comment]
Will try to do that later today. Feel free to do so yourself if I do not get to it in time.

File: core/src/main/java/hudson/FilePath.java [hunk 1]
```diff
@@ -2995,8 +2996,17 @@ private void writeObject(ObjectOutputStream oos) throws IOException {
         oos.writeBoolean(channel==null);
     }
 
+    private Channel _getChannelForSerialization() {
+        try {
+            return getChannelForSerialization();
+        } catch (NotSerializableException x) {
+            LOGGER.log(Level.WARNING, "A FilePath object is being serialized when it should not be, indicating a bug in a plugin. See https://jenkins.io/redirect/filepath-serialization for details.", x);
```
Comment (jglick) [inline core/src/main/java/hudson/FilePath.java]: https://github.com/jenkins-infra/jenkins.io/pull/2233, to begin with just pointing to the existing JIRA issue.

**Review by oleg-nenashev — APPROVED** [review]
Fine with me though we only one class is addressed. There is no known issues for other classes, so it should be ok for now

**Review by daniel-beck — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
I think we can go forward with it. The fix should prevent fatal failures in the core

**Pull request merged**